mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/SingleFileCodeGenerator.java

```diff
@@ -34,14 +35,16 @@ protected SingleFileCodeGenerator(ProcessorContext context) {
 
   @Override
   public void generate() {
-    context.getFiler().write(getFileName(), getContents());
+    ClassName typeName = getPrincipalTypeName();
+    String fileName = typeName.simpleName();
+    if (!typeName.packageName().isEmpty()) {
+      fileName = typeName.packageName() + '.' + fileName;
+    }
+    context.getFiler().write(fileName, getContents());
```

Comment:
Those changes are unrelated to this ticket.
I just noticed that the mapper crashed for types in the root package.

Reply:
Maybe we can add an integration test for a class that is in a root package? Using `MapperProcessorTest`